llm4pddl/main.py

```diff
@@ -47,9 +47,14 @@ def _run_pipeline(approach: BaseApproach, env: BaseEnv) -> None:
     """Run training and evaluation."""
     # Run training for learning-based approaches.
     if approach.is_learning_based:
-        train_tasks = env.get_train_tasks()
+        train_tasks = env.get_train_tasks()  #list of tasks
         dataset = create_dataset(train_tasks)
         approach.train(dataset)
+        # Embedding the training tasks.
```

Comment:
rather than changing main, can we have the approach do this instead? e.g., inside of `LLMOpenLoopApproach.__init__`

Reply:
good idea, except we need the `train_tasks` and `dataset` variables in order to these operations and these are created in main in the `_run_pipeline` function. How would you go about doing that in init instead?

Reply:
ah good point. so then can we do it in train() instead? we can reconstruct the train tasks from the dataset, or alter train() to also take in the train tasks.

then you will probably also need to change the call to train() that you're currently adding inside the approach itself. for that, I would recommend making a helper function, like _create_prompt_prefix(), and using that instead of train(). that makes more sense anyway because it's not really like we're re-"train"-ing the model when we solve each problem.